standard/date_ext.lua

```diff
@@ -33,15 +33,26 @@ DateExt.defaultYear = '0000'
 --- The timezone offset is incorporated into the timestamp, and the timezone is discarded.
 --- If the timezone is not specified, then the date is assumed to be in UTC.
 --- Throws if the input string is non-empty and not a valid date.
----@param dateString string|number
+---@param dateString string|number|osdate|osdateparam?
 ---@return integer?
 function DateExt.readTimestamp(dateString)
+	-- due to metatable stuff for osdate empty checks do not work properly on it
+	-- hence check if year, month and day are set instead
+	if type(dateString) == 'table' and dateString.year and dateString.month and dateString.day then
+		-- in this case we have osdate really being osdateparam
+		---@cast dateString osdateparam
+		return tonumber(os.time(dateString))
+	end
+
 	if Logic.isEmpty(dateString) then
 		return nil
 	elseif type(dateString) == 'number' then
 		return dateString
 	end
 
+	-- everything but strings was processed above
+	---@cast dateString string
```

Comment:
```suggestion
	---@cast dateString -nil
```
Think this should be good enough

Reply:
no it is not, the osdate/osdateparam would need to be casted away too:
![Screenshot 2024-03-04 130500](https://github.com/Liquipedia/Lua-Modules/assets/75081997/8f91e423-d6c7-42de-ae78-1ca1b38084b9)